http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/CbasDAO.scala

```diff
@@ -0,0 +1,12 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import cats.mtl.Ask
+import org.broadinstitute.dsde.workbench.leonardo.AppContext
+import org.http4s.Uri
+import org.http4s.headers.Authorization
+
+trait CbasDAO[F[_]] {
```

Comment:
I made `CbasDAO`, `CromwellDAO`, and `WdsDAO`. They are similar so there is some duplication, but they have slight differences (different status endpoints, payloads).

Each of these DAOs are cloud-agnostic, so can be used on Azure or GCP.

Reply:
doesn't look like this is being used for GCP? or does this not apply to GCP yet?

Reply:
Yeah it's not used for GCP but probably should (as CBAS is deployed on GCP today). I think `GKEIntepreter` today just uses https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDAO.scala to poll the root endpoint of the app, but not all individual services that might be running on the app..